managed-ledger/src/main/java/org/apache/bookkeeper/mledger/LedgerOffloader.java

```diff
@@ -22,19 +22,92 @@
 import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
-
 import org.apache.bookkeeper.client.api.ReadHandle;
 import org.apache.bookkeeper.common.annotation.InterfaceAudience;
 import org.apache.bookkeeper.common.annotation.InterfaceStability;
+import org.apache.bookkeeper.mledger.impl.EntryImpl;
+import org.apache.bookkeeper.mledger.impl.PositionImpl;
+import org.apache.bookkeeper.mledger.proto.MLDataFormats;
 import org.apache.pulsar.common.policies.data.OffloadPolicies;
 
 /**
- * Interface for offloading ledgers to long-term storage
+ * Interface for offloading ledgers to long-term storage.
  */
 @InterfaceAudience.LimitedPrivate
 @InterfaceStability.Evolving
 public interface LedgerOffloader {
 
+    interface SegmentInfo {
```

Comment:
> Is it used for managed ledger or any other component? If it only for offload internal, we don't need to expose it.

Will be used in managed ledger

Reply:
Could you provide more information about how the managed ledger uses this interface? in my opinion, the ledger offloader only create a ledgerHandle then offer entry by the ledgerHandle and get the offload result. And I think the PIP https://github.com/apache/pulsar/wiki/PIP-76%3A-Streaming-Offload also does not mention this interface. Does the offload result can't work here?

Reply:
> Could you provide more information about how the managed ledger uses this interface? in my opinion, the ledger offloader only create a ledgerHandle then offer entry by the ledgerHandle and get the offload result. And I think the PIP https://github.com/apache/pulsar/wiki/PIP-76%3A-Streaming-Offload also does not mention this interface. Does the offload result can't work here?

In my current implementation, the managed ledger will keep a reference of the SegmentInfo, and once the offload got result, it will check if the result is the same with the kept one.